### PR TITLE
Multi criteria for nested document where clause

### PR DESCRIPTION
**Pull request by clharge**
```
RelationParentModel.where(subs: {name: 'sub2', size: 2})
```
generates
```
select raw meta().id from `billeo-cb-bucket` where type = 'relation_parent_model' AND any sub in subs satisfies (sub.name = 'sub2' AND sub.size = 2) end order by meta().id
```

### CODE REVIEW

File: lib/couchbase-orm/utilities/query_helper.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ module QueryHelper
 
         module ClassMethods
 
+            #         .where(medical_folder_id: medical_folders.pluck(:id), encounter_acts: { care_plan_uuid: id, deleted_at: nil })
+
```
Comment (pimpin) [inline lib/couchbase-orm/utilities/query_helper.rb]:
```suggestion

```

File: lib/couchbase-orm/utilities/query_helper.rb [hunk 2]
```diff
@@ -12,7 +14,10 @@ def build_match(key, value)
                     "#{key} IS NULL"
                 when value.nil? && !use_is_null
                     "#{key} IS NOT VALUED"
-                when value.is_a?(Hash)
+                    # key: encounter_acts, value: { care_plan_uuid: id, deleted_at: nil })
```
Comment (pimpin) [inline lib/couchbase-orm/utilities/query_helper.rb]:
```suggestion
```

**Review by pimpin — APPROVED** [review]
Thanks for that bold and quick move.

**Review by JulienBURNET-FAUCHE — APPROVED** [review]

**Comment by JulienBURNET-FAUCHE** [comment]
The failing test seems to be flaky (order dependant)

**Pull request merged**